annotation-solr/src/main/java/eu/europeana/annotation/solr/service/impl/SolrAnnotationServiceImpl.java

```diff
@@ -589,6 +592,35 @@ private String solrUniquenessQueryLinking(Annotation anno) {
 		solrQuery+=")";
 		return solrQuery;
 	}
+	
+	private String solrUniquenessQueryLinkForContributing(Annotation anno) {
+      String solrQuery = "(" + WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.LINKFORCONTRIBUTING.getOaType() + "\"";
```

Comment:
We have to refactor these methods to build a SqlQuery objects instead of building the query String. For performance reasons the SolrQuery objects should use the targetUri as query and to add the other constraints as filterQueries

Reply:
The same refactoring needs to be applied to all other methods that build Solr queries

Reply:
Improved for all methods.